src/utils/global_reentrancy_guard.cairo

```diff
@@ -5,12 +5,21 @@
 use starknet::ContractAddress;
 
 // Satoru imports
-use satoru::data::data_store::{IDataStoreDispatcher};
+use satoru::data::data_store::{IDataStoreDispatcher, IDataStoreDispatcherTrait};
 
 /// Modifier to avoid reentrancy.
-fn non_reentrant_before(data_store: IDataStoreDispatcher) { // TODO
+fn non_reentrant_before(data_store: IDataStoreDispatcher) { 
+    let status = match data_store.get_bool('REENTRANCY_GUARD_STATUS') {
+        Option::Some(v) => v,
+        Option::None(()) => false,
+    }; // Read key from Data Store
+
+    assert(!status, 'ReentrancyGuard: reentrant call');
```

Comment:
nit: error message in a constant in an error file, following conventions of other error messages